plugin/procfs/procfs.cc

```diff
@@ -0,0 +1,527 @@
+    std::string path;
+    std::getline(list, path, ';');
+    if (path.rfind("/proc", 0) != 0 && path.rfind("/sys", 0) != 0) continue;
+
+    if (path.find_first_of("*?[") == std::string::npos) {
+      files.push_back(path);
+      continue;
+    }
+
+    std::vector<std::string> v1 =
+        limited_glob(path, MAX_FILES_PER_PATTERN, MAX_PATTERN_DEPTH);
+    files.insert(files.end(), v1.begin(), v1.end());
+  }
+}
+
+static void fill_view_row(THD *thd, TABLE *table, const char *fname, char *buf,
+                          size_t sz) {
+  if (sz == 0) return;
+
+  table->field[0]->store(fname, strlen(fname), system_charset_info);
+  table->field[1]->store(buf, sz, system_charset_info);
+  schema_table_store_record(thd, table);
+
+  files_read++;
+  bytes_read += sz;
+}
+
+static int fill_view(THD *thd, TABLE_LIST *tables, Item *cond) {
+  if (!thd->security_context()
+           ->has_global_grant(PRIVILEGE_NAME, strlen(PRIVILEGE_NAME))
+           .first) {
+    my_error(ER_SPECIFIC_ACCESS_DENIED_ERROR, MYF(0), PRIVILEGE_NAME);
+    access_violations++;
+    return 1;
+  }
+
+  TABLE *table = tables->table;
+
+  static const std::string I_S_FILE("INFORMATION_SCHEMA.PROCFS.FILE");
+  std::string like_arg;
+
+  std::map<std::string, bool> in_args;
+
+  if (cond != 0) {
+    std::string eq_arg;
+    if (!get_equal_condition_argument(cond, &eq_arg, I_S_FILE) &&
+        !eq_arg.empty()) {
+      in_args[eq_arg] = true;
+    } else if (!get_like_condition_argument(cond, &like_arg, I_S_FILE) &&
+               !like_arg.empty()) {
+    } else {
+      get_in_condition_argument(cond, in_args, I_S_FILE);
+    }
+  }
+
+  std::vector<std::string> files;
+  fill_files_list(files);
+  for (std::vector<std::string>::const_iterator fname = files.begin();
+       fname != files.end(); ++fname) {
+    if (cond != 0 && in_args.size() > 0 &&
+        in_args.find(*fname) == in_args.end())
+      continue;
+    if (cond != 0 && like_arg.size() > 0 &&
+        wild_compare(fname->c_str(), fname->size(), like_arg.c_str(),
+                     like_arg.size(), 0))
+      continue;
+
+    std::ifstream f(fname->c_str());
+    if (!f || !f.is_open()) continue;
+
+    f.read(buffer, MAX_CONTENTS_SIZE);
+    fill_view_row(thd, table, fname->c_str(), buffer, f.gcount());
+    f.close();
+  }
+
+  queries++;
+
+  return false;
+}
+
+static int view_init(void *ptr) {
+  if (init_logging_service_for_plugin(&reg_srv, &log_bi, &log_bs)) {
+    return 1;
+  }
+
+  LogErr(INFORMATION_LEVEL, ER_LOG_PRINTF_MSG, "Plugin procfs initializing...");
+
+  if (strcmp(files_spec, DEFAULT_FILES_SPEC) != 0) {
+    LogPluginErrMsg(WARNING_LEVEL, ER_LOG_PRINTF_MSG,
+                    "default procfs_files_spec has been overridden with "
+                    "\"%s\", sensitive system information may be exposed via "
+                    "the information_schema.procfs table",
+                    files_spec);
+  }
+
+  SERVICE_TYPE(registry) *reg = mysql_plugin_registry_acquire();
```

Comment:
At this point plugin registry has already been acquired by the `init_logging_service_for_plugin()` call - you can simply use `reg_srv` here.

Reply:
Ahh, right, the example I got this from a structured this way but I see that these are the same type and interchangeable, that simplifies things greatly.